packages/flutter/lib/src/material/page.dart

```diff
@@ -64,9 +124,6 @@ class MaterialPageRoute<T> extends PageRoute<T> {
   Color get barrierColor => null;
 
   @override
-  bool canTransitionFrom(TransitionRoute<dynamic> nextRoute) => false;
```

Comment:
You want to return true in the specific case of transitioning from one route that supports this to another that supports this, but you don't want to return true across the board. To see why, put a hero on a page then show a dialog with a matching hero. If I'm not mistaken, if this returns true, it'll try to animate from one to the other. Or indeed, put your iOS-style route before a dialog route, same thing without heroes.